disco_aws_automation/disco_rds.py

```diff
@@ -79,17 +79,25 @@ def config_truthy(self, section, param, default='True'):
         """Read the RDS config file and extract the boolean value, or return default if missing"""
         return is_truthy(self.config_with_default(section, param, default))
 
-    def get_master_password(self, instance_identifier):
+    def get_master_password(self, env_name, database_name):
         """
         Get the Master Password for instance stored in the S3 bucket
         """
-        s3_password_key = 'rds/{0}/master_user_password'.format(instance_identifier)
         bucket_name = self.vpc.get_credential_buckets_from_env_name(self.config_aws, self.vpc_name)[0]
-        return DiscoS3Bucket(bucket_name).get_key(s3_password_key)
 
-    def get_instance_parameters(self, instance_identifier):
+        try:
+            # for backwards compatibility check the old style keys containing the env name
+            instance_identifier = self._get_instance_identifier(env_name, database_name)
+            s3_password_key = 'rds/{0}/master_user_password'.format(instance_identifier)
+            return DiscoS3Bucket(bucket_name).get_key(s3_password_key)
+        except KeyError:
+            # if a key doesn't exist for the database identifier then try the database name
+            s3_password_key = 'rds/{0}/master_user_password'.format(database_name)
+            return DiscoS3Bucket(bucket_name).get_key(s3_password_key)
```

Comment:
Isn't there a has_key or something like that?  Or does that just do more API calls than we need to?

I should probably find a way to migrate the S3passwordmanager from asiaq_db_manager into here, it's ... similar.